python/lsst/resources/http.py

```diff
@@ -531,49 +580,269 @@ def transfer_from(
                 src.remove()
 
-    def _do_put(self, data: Union[BinaryIO, bytes]) -> None:
-        """Perform an HTTP PUT request taking into account redirection."""
-        final_url = self.geturl()
-        if _SEND_EXPECT_HEADER_ON_PUT:
-            # Do a PUT request with an empty body and retrieve the final
-            # destination URL returned by the server.
-            headers = {"Content-Length": "0", "Expect": "100-continue"}
-            resp = self.put_session.put(
-                final_url, data=None, headers=headers, allow_redirects=False, timeout=TIMEOUT
+    def _as_local(self) -> Tuple[str, bool]:
+        """Download object over HTTP and place in temporary directory.
+
+        Returns
+        -------
+        path : `str`
+            Path to local temporary file.
+        temporary : `bool`
+            Always returns `True`. This is always a temporary file.
+        """
+        resp = self.session.get(self.geturl(), stream=True, timeout=TIMEOUT)
+        if resp.status_code != requests.codes.ok:
+            raise FileNotFoundError(f"Unable to download resource {self}; status code: {resp.status_code}")
+
+        tmpdir, buffering = _get_temp_dir()
+        with tempfile.NamedTemporaryFile(
+            suffix=self.getExtension(), buffering=buffering, dir=tmpdir, delete=False
+        ) as tmpFile:
+            with time_this(
+                log,
+                msg="Downloading %s [length=%s] to local file %s [chunk_size=%d]",
+                args=(self, resp.headers.get("Content-Length"), tmpFile.name, buffering),
+            ):
+                for chunk in resp.iter_content(chunk_size=buffering):
+                    tmpFile.write(chunk)
+
+        return tmpFile.name, True
+
+    def _send_webdav_request(
+        self, method: str, url: Optional[str] = None, headers: dict[str, str] = {}, body: Optional[str] = None
+    ) -> requests.Response:
+        """Send a webDAV request and correctly handle redirects.
+
+        Parameters
+        ----------
+        method : `str`
+            The mthod of the HTTP request to be sent, e.g. PROPFIND, MKCOL.
+        headers : `dict`, optional
+            A dictionary of key-value pairs (both strings) to include as
+            headers in the request.
+        body: `str`, optional
+            The body of the request.
+
+        Notes
+        -----
+        This way of sending webDAV requests is necessary for handling
+        redirection ourselves, since the 'requests' package changes the method
+        of the redirected request when the server responds with status 302 and
+        the method of the original request is not HEAD (which is the case for
+        webDAV requests).
+
+        That means that when the webDAV server we interact with responds with
+        a redirection to a PROPFIND or MKCOL request, the request gets
+        converted to a GET request when sent to the redirected location.
+
+        See `SessionRedirectMixin.rebuild_method()` in
```

Comment:
I think you need 
```
`requests.sessions.SessionRedirectMixin.rebuild_method()`
```
ie the full path. Even then it might depend on sphinx knowing where the requests documentation is located.

Reply:
Fixed.